src/figuresClasses.ts

```diff
@@ -1,19 +1,69 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape= 'triangle';
 
-}
+  constructor(public color: Color, public a: number,
+    public b: number, public c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('length has to be a positive number');
+    }
+
+    if (!this.isValidTriangle()) {
+      throw new Error('highest side needs to be lower than sum of other sides');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
-export class Circle {
+    return Math.floor(area * 100) / 100;
+  }
 
+  private isValidTriangle(): boolean {
+    const [a, b, c] = [this.a, this.b, this.c].sort((x, y) => x - y);
+
+    return a + b > c;
+  }
```

Comment:
Maybe lets create a method called for example "getSortedSides" and make it return the sides sorted in ascending order?

```suggestion
  private isValidTriangle(): boolean {
    const [a, b, c] = this.getSortedSides();
    return a + b > c;
  } 
```